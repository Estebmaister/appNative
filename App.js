import { StatusBar } from "expo-status-bar";
import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { color } from "react-native-reanimated";

export default function App() {
  return (
    <View style={styles.container}>
      <Text style={{ color: "#eee" }}>
        Open up App.js to start working on your app!
      </Text>

      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#223",
    alignItems: "center",
    justifyContent: "center",
  },
});
