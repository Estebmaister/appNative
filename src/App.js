import { StatusBar } from "expo-status-bar";
import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { color } from "react-native-reanimated";

export default function App() {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>Logo</Text>
      <Text style={styles.text}>Hello World from app.js!</Text>
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#214192",
    alignItems: "center",
    justifyContent: "center",
  },
  title: {
    position: "absolute",
    top: 100,
    paddingVertical: 40,
    borderWidth: 4,
    borderColor: "#223",
    borderRadius: 100,
    width: 150,
    height: 150,
    backgroundColor: "#00ff8f",
    color: "#223",
    textAlign: "center",
    fontSize: 40,
    fontWeight: "bold",
  },
  text: {
    color: "#eee",
    fontSize: 30,
  },
});
