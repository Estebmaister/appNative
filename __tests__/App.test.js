import "react-native";
import React from "react";
import App from "../src/App";
import Index from "../index.js";
import Metro from "../metro.config.js";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

describe("<Index />", () => {
  it("Is not modify", () => {
    expect(Index).toEqual({});
  });
});

describe("<Metro />", () => {
  it("Is not modify", () => {
    expect(Metro).toEqual({
      transformer: {
        assetPlugins: ["expo-asset/tools/hashAssetFiles"],
      },
    });
  });
});

describe("<App />", () => {
  it("Renders", () => {
    renderer.create(<App />);
  });

  it("Has 2 children", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(2);
  });

  it("Renders correctly to prev snap", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
