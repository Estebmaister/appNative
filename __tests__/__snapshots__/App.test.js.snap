// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#214192",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#00ff8f",
        "borderColor": "#223",
        "borderRadius": 100,
        "borderWidth": 4,
        "color": "#223",
        "fontSize": 40,
        "fontWeight": "bold",
        "height": 150,
        "paddingVertical": 40,
        "position": "absolute",
        "textAlign": "center",
        "top": 100,
        "width": 150,
      }
    }
  >
    Logo
  </Text>
  <Text
    style={
      Object {
        "color": "#eee",
        "fontSize": 30,
      }
    }
  >
    Hello World from app.js!
  </Text>
</View>
`;
